app/helpers/application_helper/button/basic.rb

```diff
@@ -12,6 +14,11 @@ def initialize(view_context, view_binding, instance_data, props)
     end
   end
 
+  def role_allows_feature?
+    return true if id.nil?
+    role_allows?(:feature => id)
```

Comment:
So `id` is defined as `self[:child_id]` and that is set only for buttons under a drop-down.

So if we have this in the base class then for most buttons `id` will be `nil` and therefor the method will return `true`.

This looks very wrong.

Maybe we should finally split button's id (RBAC feature) and DOM id (that is composed from parent id and child id) to avoid this confusion.